src/energyplus/ForwardTranslator/ForwardTranslateCoilSystemCoolingWater.cpp

```diff
@@ -177,20 +173,21 @@ namespace energyplus {
     idfObject.setDouble(CoilSystem_Cooling_WaterFields::MinimumWaterLoopTemperatureForHeatRecovery, minimumWaterLoopTemperatureForHeatRecovery);
 
     // Companion Coil Used For Heat Recovery: Optional Object
-    if (boost::optional<WaterToAirComponent> companionCoilUsedForHeatRecovery_ = modelObject.companionCoilUsedForHeatRecovery()) {
-      if (boost::optional<IdfObject> wo_ = translateAndMapModelObject(companionCoilUsedForHeatRecovery_.get())) {
-        idfObject.setString(CoilSystem_Cooling_WaterFields::CompanionCoilUsedForHeatRecovery, wo_->nameString());
-        if (wo_->iddObject().type() == IddObjectType::Coil_Cooling_Water) {
-          wo_->setString(Coil_Cooling_WaterFields::AirInletNodeName, modelObject.airLoopHVACOutdoorAirSystem()->reliefAirModelObject()->nameString());
-          wo_->setString(Coil_Cooling_WaterFields::AirOutletNodeName, wo_->nameString() + " Exhaust Outlet Node");  // FIXME
-          // Add IddObjectType::Coil_Cooling_Water_DetailedGeometry if implemented
+    if (boost::optional<HVACComponent> companionCoilUsedForHeatRecovery_ = modelObject.companionCoilUsedForHeatRecovery()) {
+      if (auto comp_ = companionCoilUsedForHeatRecovery_->optionalCast<CoilCoolingWater>()) {
+        if (comp_->airInletModelObject()) {
+          if (boost::optional<IdfObject> idf_ = translateAndMapModelObject(companionCoilUsedForHeatRecovery_.get())) {
+            idfObject.setString(CoilSystem_Cooling_WaterFields::CompanionCoilUsedForHeatRecovery, idf_->nameString());
+          }
         } else {
-          // Shouldn't happen, accepts only Coil:Cooling:Water or Coil:Cooling:Water:DetailedGeometry
-          // Shouldn't happen, accepts only Coil:Cooling:DX:SingleSpeed or Coil:Cooling:DX:VariableSpeed
-          LOG(Fatal, modelObject.briefDescription() << " appears to have a companion coil used for heat recovery that shouldn't have been accepted: "
-                                                    << companionCoilUsedForHeatRecovery_.get().briefDescription());
-          OS_ASSERT(false);
+          LOG(Warn, modelObject.briefDescription()
+                      << " has a companion coil used for heat recovery that is not connected to an AirLoopHVAC / AirLoopHVACOutdoorAirSystem"
+                      << " and will not be translated:" << comp_->briefDescription());
         }
+      } else {
+        // Shouldn't happen, accepts only Coil:Cooling:Water
+        LOG_AND_THROW(modelObject.briefDescription() << " appears to have a companion coil used for heat recovery that shouldn't have been accepted: "
+                                                     << companionCoilUsedForHeatRecovery_.get().briefDescription());
```

Comment:
No translating the companion coil if it's not connected by itself on an AirLoopHVAC/AirLoopHVACOutdoorAIrSystem